idaes/core/base/property_base.py

```diff
@@ -505,44 +505,63 @@ def report(self, index=(0), true_state=False, dof=False, ostream=None, prefix=""
 
         ostream.write("\n" + "=" * max_str_length + "\n")
 
-    def build_port(self, target_block, port_name, doc=None, subset=None):
+    def get_port_reference_name(self, component_name, port_name):
+        """
+        Get the standard name of a "port reference", the component
+        accessed on the port by "port_name.component_name".
+
+        """
+        return "_%s_%s_ref" % (component_name, port_name)
+
+    def build_port(
+        self,
+        target_block,
+        port_name,
+        doc=None,
+        slice_index=None,
+        index=None,
+    ):
         """
         Constructs a Port based on this StateBlock attached to the target block.
 
         Args:
             target_block - block to which Port should be attached
             port_name - name to use for Port
             doc - doc string or Prot object
-            subset - slicer representing a subset of indices of StateBlock which
-                should be included in Port
+            slice_index - Slice index (e.g. (slice(None), 0.0) that will be
+                used to index self when constructing port references.
```

Comment:
In addition to being easier to name the optional argument, I think specifying an index here is safer. By specifying a slice, I think we are implicitly assuming that the slice refers to a component where `self`'s port members are valid components (typically just a slice of self). By specifying a slice index then slicing `self` explicitly, we know that the port members are valid components (unless `self` has implemented a bad `define_port_members` method...)

Reply:
Accessing an index with a colon, e.g. `obj[:,0.0]`, is the same as accessing with a slice object, e.g. `obj[slice(None), 0.0]`. The difference is that a colon is not valid syntax in a parenthesized tuple, so a slice object must be used instead. The construction arguments of the slice are `slice(start, stop, step)`, which are the same as indexing an object `obj[start:stop:step]` in colon-syntax. For some reason, you can't just create a slice like `slice()` to represent a basic slice, you need to provide at least one argument of `None`. Note:
```python
>>> slice(None) == slice(None, None, None)
True
```